src/main/kotlin/io/perfometer/http/client/HttpConnectionBuilder.kt

```diff
@@ -42,8 +42,10 @@ internal class HttpConnectionBuilder(
         this.connection.requestMethod = method
     }
 
-    fun headers(headers: Map<String, String>) {
-        headers.forEach { this.connection.setRequestProperty(it.key, it.value) }
+    fun headers(headers: Map<String, List<String>>) {
+        headers
+            .flatMap { entry -> entry.value.map { Pair(entry.key, it) } }
+            .forEach { this.connection.addRequestProperty(it.first, it.second) }
```

Comment:
I'd prefer to have the notation as you've used in KtorHttpClient, so `name, value -> this.connection.addRequestProperty(name, value)` instead of implicit `it`.

Reply:
done